### PR TITLE
Add --direct-only flag to only work on direct dependencies

### LINKED ISSUE #2: Flag to only report *direct* dependencies
I think it will be a good idea if `modtools check` (but also probably `modtools update`) work only on *direct* dependencies instead of all of them. 

If there is any interest, and we can agree on the flag I can try to add it?

### PR DESCRIPTION
**Pull request by mstoykov**
closes #2

### CODE REVIEW

**Comment by mstoykov** [comment]
I would've preferred if I could add a test, but it seems like that will be quite complicated and fragile. I will need to have a project with dependencies, some up for update some not and this needs to stay that way forever. Don't know if you haven't thought on it and have found some solution?

File: cmd/check.go [hunk 1]
```diff
@@ -14,7 +15,8 @@ func init() {
 		Short: "Check for out-of-date dependencies",
 		Long:  "Scans through direct and indirect dependencies to check if a newer version is available. Exceptions can be set by 'modtools freeze'",
 		RunE: func(cmd *coral.Command, args []string) error {
-			res, err := checkDeps()
+			err := rootCmd.Flags().Parse(os.Args)
```
Comment (dop251) [inline cmd/check.go]: This value of `err` is never used. Also, I think this is not needed at all as coral parses the flags automatically.

**Comment by dop251** [comment]
> 

I think the easiest way would be to decouple `checkDeps()` from `loadExceptions()` and `readDeps()`, i.e. have `checkDeps()` take the results of these as parameters. This would allow writing unit tests for checkDeps() logic. Same approach could be applied to readDeps(), but frankly I don't consider either a top priority for now.

File: cmd/check.go (hunk 1 above)
Comment (mstoykov) [reply cmd/check.go]: yeah that is true, I was trying  to figure out why something wasn't working and then didn't drop this line :facepalm:

File: cmd/modtools.go [hunk 2]
```diff
@@ -6,17 +6,20 @@ import (
...
-)
+var rootCmd = &coral.Command{
+	Use: "modtools",
+	PersistentPreRun: func(cmd *coral.Command, args []string) {
+		cmd.SilenceUsage = true
+	},
+}
 
 func Execute() {
+	rootCmd.PersistentFlags().Bool("directOnly", false, "work only on direct dependencies")
```
Comment (mstoykov) [inline cmd/modtools.go]: Maybe this should be  `--direct-only` :thinking:

Comment (dop251) [reply cmd/modtools.go]: Yes

Comment (mstoykov) [reply cmd/modtools.go]: fixed and updated the message and the PR title

**Pull request merged**